### ISSUE TITLE
pip-sync removes pkg-resoources

### ISSUE THREAD

**Issue by graingert** [jazzband/pip-tools#389]
Describe the issue briefly here.
##### Steps to replicate

``` shell
$ cat requirements-dev.txt 
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file requirements-dev.txt requirements-dev.in
#
click==6.6                # via pip-tools
first==2.0.1              # via pip-tools
mypy-lang==0.4.4
pip-tools==1.7.0
pluggy==0.3.1             # via tox
py==1.4.31                # via tox
six==1.10.0               # via pip-tools
tox==2.3.1
virtualenv==15.0.3        # via tox
wheel==0.29.0

# The following packages are commented out because they are
# considered to be unsafe in a requirements file:
# pip
# setuptools
```
##### Expected result

``` shell
```
##### Actual result

``` shell
$ pip-sync requirements-dev.txt 
Uninstalling pkg-resources-0.0.0:
  Successfully uninstalled pkg-resources-0.0.0
```

**Comment by graingert** [comment]
Now I can't install any packages

**Comment by das-g** [comment]
I noticed this, too. I guess this is probably due to (or for the same reason as) pypa/pip#4022, which according to @RonnyPfannschmidt is caused by a bug in ubuntu. (Filed as https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1635463)

**Comment by estan** [comment]
Our CI builds recently started failing because of this. Since it seems the Ubuntu bug is slow-moving, would it be OK to add @graingert's [workaround](https://github.com/graingert/pip-tools/commit/1212c172203b643978b4dc18399e6f345b798abd) to official piptools? It just adds `pkg-resources` to `PACKAGES_TO_IGNORE` in `sync.py`.

**Comment by das-g** [comment]
Does anyone know what would have to change on the Ubuntu side to fix this properly? Can we send them a patch to speed things up?

Until then, an official pip-tools release with the workaround merged would be much appreciated.

**Comment by vphilippon** [comment]
@das-g The bad news is, we're currently blocked and can't do any official release of `pip-tools` (see #531).

As soon as we're able to make new releases, I'll check to add that package to the list, unless Ubuntu manages to fix that first.

**Comment by vphilippon** [comment]
Closed by #555
